org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/junit/JUnitFragment.xtend

```diff
@@ -147,7 +157,54 @@ class JUnitFragment extends AbstractStubGeneratingFragment {
 			}
 		''')
 	}
-	
+
+	def protected JavaFileAccess generateJavaExampleRuntimeTest() {
+		val xtextRunner = new TypeReference(testingPackage + ".XtextRunner")
+		val runWith = new TypeReference("org.junit.runner.RunWith")
+		val extendWith = new TypeReference("org.junit.jupiter.api.extension.ExtendWith")
+		val injectionExtension = new TypeReference("org.eclipse.xtext.testing.extensions.InjectionExtension")
+		val injectWith = new TypeReference(testingPackage + ".InjectWith")
+		val parseHelper = new TypeReference(testingPackage + ".util.ParseHelper")
+		val test = switch (junitVersion) {
+			case JUnitVersion.JUNIT_4: new TypeReference("org.junit.Test")
+			case JUnitVersion.JUNIT_5: new TypeReference("org.junit.jupiter.api.Test")
+		}
+		val assert = switch (junitVersion) {
+			case JUnitVersion.JUNIT_4: new TypeReference("org.junit.Assert")
+			case JUnitVersion.JUNIT_5: new TypeReference("org.junit.jupiter.api.Assertions")
+		}
+		val rootType = new TypeReference(grammar.rules.head.type.classifier as EClass, grammar.eResource.resourceSet)
+		val list = new TypeReference("java.util.List")
+		val diagnostic = new TypeReference("org.eclipse.emf.ecore.resource", "Resource.Diagnostic")
+		val iterableExtensions = new TypeReference("org.eclipse.xtext.xbase.lib.IterableExtensions")
+		return fileAccessFactory.createJavaFile(exampleRuntimeTest, '''
+			�IF junitVersion==JUnitVersion.JUNIT_4�
+				@�runWith�(�xtextRunner�.class)
+			�ENDIF�
+			�IF junitVersion==JUnitVersion.JUNIT_5�
+				@�extendWith�(�injectionExtension�.class)
+			�ENDIF�
+			@�injectWith�(�injectorProvider�.class)
+			public class �exampleRuntimeTest� {
+				@�Inject�
+				private �parseHelper�<�rootType�> parseHelper;
+				
+				@�test�
+				public void loadModel() throws Exception {
+					Model result = parseHelper.parse("Hello Xtext!");
+					�assert�.assertNotNull(result);
+					�list�<�diagnostic�> errors = result.eResource().getErrors();
+					�IF junitVersion==JUnitVersion.JUNIT_4�
+						�assert�.assertTrue("Unexpected errors: " + �iterableExtensions�.join(errors, ", "), errors.isEmpty());
```

Comment:
Shall we create `Strings.join(..)` instead?

Reply:
Please ignore this suggestion.

Reply:
I am fine. The only downside is the added verbosity. I'll make the change if it is acceptable.
```
«list»<«diagnostic»> diagnostics = result.eResource().getErrors();
CharSequence[] errors = diagnostics.stream().map(Object::toString).toArray(CharSequence[] :: new);
```

Reply:
No, please don't go down that path. `IterableExtensions.join(..)` is a good fit here.